mathesar/utils/tables.py

```diff
@@ -29,3 +34,15 @@ def get_table_column_types(table):
         and not col.foreign_keys
     }
     return col_types
+
+
+def update_sa_table(table, validated_data):
+    for arg in validated_data:
+        if arg not in SUPPORTED_TABLE_UPDATE_ARGS:
+            raise ValidationError({arg: f'Updating {arg} for tables is not supported.'})
+    if 'name' in validated_data:
+        rename_table(table.name, table.schema.name, table.schema._sa_engine,
+                     validated_data['name'])
+        # Force the table and nameto reload
+        del table._sa_table
+        del table.name
```

Comment:
I think we should either rework this function to only validate, then call an `update_table` in the `db.tables` module, or move the `update_column` from the `db.columns` module to a new `mathesar.utils.columns` module.  My preference is the former, since I think that `update_table` function will be useful outside of the Mathesar webservice.

Reply:
I agree with choosing the former.